examples/fluids/src/setupstg_shur14.c

```diff
@@ -0,0 +1,333 @@
+    }
+  } else {
+    printf("%s does not contain shape of contained data in the first line", path);
+    PetscFunctionReturn(-1);
+  }
+  ierr = PetscStrToArrayDestroy(ndims, array); CHKERRQ(ierr);
+
+  CeedScalar *phi = &stg_ctx->data[stg_ctx->offsets.phi];
+  CeedScalar (*d)[stg_ctx->nmodes]     = (CeedScalar (*)[stg_ctx->nmodes])
+                                         &stg_ctx->data[stg_ctx->offsets.d];
+  CeedScalar (*sigma)[stg_ctx->nmodes] = (CeedScalar (*)[stg_ctx->nmodes])
+                                         &stg_ctx->data[stg_ctx->offsets.sigma];
+
+  for (int i=0; i<stg_ctx->nmodes; i++) {
+    if ( fgets(line, char_array_len, fp) != NULL) {
+      ierr = PetscStrToArray(line, ' ', &ndims, &array); CHKERRQ(ierr);
+      if(ndims < dims[1]) {
+        printf("Line %d of %s does not contain enough columns (%d instead of %d)", i,
+               path, ndims, dims[1]);
+        PetscFunctionReturn(-1);
+      }
+      d[0][i]     = (CeedScalar) atof(array[0]);
+      d[1][i]     = (CeedScalar) atof(array[1]);
+      d[2][i]     = (CeedScalar) atof(array[2]);
+      phi[i]      = (CeedScalar) atof(array[3]);
+      sigma[0][i] = (CeedScalar) atof(array[4]);
+      sigma[1][i] = (CeedScalar) atof(array[5]);
+      sigma[2][i] = (CeedScalar) atof(array[6]);
+    } else {
+      printf("Error reading line %d in %s", i, path);
+      PetscFunctionReturn(-1);
+    }
+  }
+  ierr = PetscFClose(comm, fp);
+  PetscFunctionReturn(0);
+}
+
+
+PetscErrorCode SetupSTGContext(STGShur14Context stg_ctx) {
+  PetscErrorCode ierr;
+
+  // Get paths for files
+  char stg_inflow_path[PETSC_MAX_PATH_LEN] = "./STGInflow.dat";
+  char stg_rand_path[PETSC_MAX_PATH_LEN] = "./STGRand.dat";
+  ierr = PetscOptionsGetString(NULL, NULL, "-stg_inflow_path", stg_inflow_path,
+                               sizeof(stg_inflow_path), NULL); CHKERRQ(ierr);
+  ierr = PetscOptionsGetString(NULL, NULL, "-stg_rand_path", stg_rand_path,
+                               sizeof(stg_rand_path), NULL); CHKERRQ(ierr);
+
+  int nmodes, nprofs;
+  GetNRows(stg_rand_path, &nmodes);
+  GetNRows(stg_inflow_path, &nprofs);
+
+  {
+    STGShur14Context s;
+    ierr = PetscCalloc1(1, &s); CHKERRQ(ierr);
+    s->nmodes = nmodes;
+    s->nprofs = nprofs;
+    s->offsets.sigma   = 0;
+    s->offsets.d       = nmodes*3;
+    s->offsets.phi     = s->offsets.d       + nmodes*3;
+    s->offsets.kappa   = s->offsets.phi     + nmodes;
+    s->offsets.prof_dw = s->offsets.kappa   + nmodes;
+    s->offsets.ubar    = s->offsets.prof_dw + nprofs;
+    s->offsets.cij     = s->offsets.ubar    + nprofs*3;
+    s->offsets.eps     = s->offsets.cij     + nprofs*6;
+    s->offsets.lt      = s->offsets.eps     + nprofs;
+    int total_num_scalars = s->offsets.lt + nprofs;
+    stg_ctx = malloc(sizeof(*stg_ctx) + total_num_scalars*sizeof(stg_ctx->data[0]));
+    *stg_ctx = *s;
+  }
+
+  stg_ctx->alpha = 1.01;
+  ierr = PetscOptionsGetReal(NULL, NULL, "-stg_alpha", &stg_ctx->alpha, NULL);
```

Comment:
`PetscOptionsBegin` and `PetscOptionsReal` would allow these options to be displayed in `-help` output.

Reply:
Ah, didn't realize that. I was just going off the PETSc user manual, which only mentions the `PetscOptionsGet*` functions. Does it also pass the default value through to `-help`?

Reply:
Yeah, `-help` will show the default at the time that code is reached. We should encourage people to use these in the users manual.